tephra-core/src/test/java/co/cask/tephra/TransactionAdminTest.java

```diff
@@ -0,0 +1,195 @@
+          }
+        }),
+      new TransactionClientModule()
+    );
+
+    zkClientService = injector.getInstance(ZKClientService.class);
+    zkClientService.startAndWait();
+
+    // start a tx server
+    txService = injector.getInstance(TransactionService.class);
+    txClient = injector.getInstance(TransactionSystemClient.class);
+    try {
+      LOG.info("Starting transaction service");
+      txService.startAndWait();
+    } catch (Exception e) {
+      LOG.error("Failed to start service: ", e);
+    }
+  }
+
+  @Before
+  public void reset() throws Exception {
+    txClient.resetState();
+  }
+
+  @AfterClass
+  public static void stop() throws Exception {
+    txService.stopAndWait();
+    zkClientService.stopAndWait();
+    zkServer.stopAndWait();
+  }
+
+  @Test
+  public void testPrintUsage() throws Exception {
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ByteArrayOutputStream err = new ByteArrayOutputStream();
+    TransactionAdmin txAdmin = new TransactionAdmin(new PrintStream(out), new PrintStream(err));
+    txAdmin.doMain(new String[0], conf);
+    //noinspection ConstantConditions
+    Assert.assertTrue(Bytes.toString(err.toByteArray()).startsWith("Usage:"));
+    Assert.assertEquals(0, out.toByteArray().length);
+  }
+
+  @Test
+  public void testTruncateInvalidTx() throws Exception {
+    Transaction tx1 = txClient.startLong();
+    Transaction tx2 = txClient.startShort();
+    txClient.invalidate(tx1.getWritePointer());
+    txClient.invalidate(tx2.getWritePointer());
+    
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ByteArrayOutputStream err = new ByteArrayOutputStream();
+    TransactionAdmin txAdmin = new TransactionAdmin(new PrintStream(out), new PrintStream(err));
+    txAdmin.doMain(new String[]{"--truncate-invalid-tx", String.valueOf(tx2.getWritePointer())}, conf);
+    List<String> expected = ImmutableList.of("Invalid list size before truncation: 2",
+                                             "Invalid list size after truncation: 1",
+                                             "");
+    Assert.assertEquals(Joiner.on("\n").join(expected), Bytes.toString(out.toByteArray()));
+    Assert.assertEquals(0, err.toByteArray().length);
+  }
+
+  @Test
+  public void testTruncateInvalidTxBefore() throws Exception {
+    Transaction tx1 = txClient.startLong();
+    TimeUnit.SECONDS.sleep(1);
+    long beforeTx2 = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+    Transaction tx2 = txClient.startLong();
+    
+    // Try before invalidation
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ByteArrayOutputStream err = new ByteArrayOutputStream();
+    TransactionAdmin txAdmin = new TransactionAdmin(new PrintStream(out), new PrintStream(err));
+    txAdmin.doMain(new String[] {"--truncate-invalid-tx-before", String.valueOf(beforeTx2)}, conf);
+    List<String> expectedOut = ImmutableList.of("Invalid list size before truncation: 0",
```

Comment:
Same as above, would prefer that we assert on the invalid list size (obtained from txClient) instead of on printed output.